packages/perseus-core/src/widgets/matcher/matcher-util.ts

```diff
@@ -0,0 +1,61 @@
+import {CoreUtil} from "@khanacademy/perseus-core";
+
+import type {PerseusMatcherWidgetOptions} from "@khanacademy/perseus-core";
+
+type MatcherInfo = {
+    left: ReadonlyArray<string>;
+    right: ReadonlyArray<string>;
+    orderMatters: boolean;
+    problemNum: number | null | undefined;
+};
+
+export const matcherShuffle = (
+    props: MatcherInfo,
```

Comment:
I could instead update to use the original widget options type, but it seemed like specifically listing which items are needed would be more useful, but I could change to be the below code instead.
```
props: PerseusMatcherWidgetOptions & {
        problemNum: number | null | undefined;
    }
```

Reply:
I think what you have is okay! I might lean toward passing the `seed` as a separate argument instead of adding a `problemNum` field to the options object.

```
export const matcherShuffle = (
  matcherOptions: PerseusMatcherWidgetOptions,
  seed: number,
) => // ...
```

Naming the parameter `problemNum` makes it seem like it has something inherently to do with the `problemNum` passed to the Renderer. Really, though, it's just the seed for the random number generator, which can be any arbitrary number (as evidenced by the rubric stripping code where you hardcode it). `matcherShuffle` is only concerned with shuffling, so its parameters should be at that level of abstraction.

Reply:
Hmm, this logic was pulled directly from the widget. If I change it to use a seed, then I would end up changing the widget's logic. Inside the widget, it is using the problemNum. When used with the public widget options logic, it wouldn't be though. I think if we keep this shuffle logic pulled out, we probably shouldn't change it.